### ISSUE TITLE
Found word is not highlighted

### ISSUE THREAD

**Issue by Pathologic** [parpalak/rose#25]
![image](https://github.com/parpalak/rose/assets/3012304/2cbc81f3-bc9d-4738-8bc2-99c04a19a16f)

It's hard for me to debug your code, but it seems that "gallery" transforms to "galleri", which is not highlighted because there's no "galleri" in the snippet text.

**Comment by parpalak** [comment]
The reason for this behavior is that Rose operates with morphology through stemmers. The idea is that one word is considered a variation of another if they have a common stem. I use heuristic Porter stemmers, specifically the English variant from here: https://github.com/markfullmer/porter2. This variant transforms 'gallery' into 'galleri.' I will explore the algorithm further to understand potential solutions. As a temporary workaround, I suggest excluding 'PorterStemmerEnglish' from your code and keeping only 'PorterStemmerRussian.' This will result in the loss of morphology support for English words, but it might be beneficial since your content is primarily in Russian.

**Comment by Pathologic** [comment]
Maybe it would be better to keep original forms for the highlighter ("gallery", "galleri")?

**Comment by parpalak** [comment]
I've found a solution that I'm really proud of: https://github.com/parpalak/rose/commit/a9acae77ce267db6d11270bbf98b4ac190076e0a

Just like in this case: https://parpalak.com/blog/2023/01/21/master_of_crutches

**Comment by parpalak** [comment]
Merged everything to the main branch.

**Comment by Pathologic** [comment]
I've updated rose and recreated index but nothing changed.

**Comment by Pathologic** [comment]
Now it highlights "galleries" but not "gallery".

**Comment by parpalak** [comment]
What is the minimun fragment of code in which the problem is reproduced?
As you can see from the unit tests, "Gallery" is highlighted as expected:

```
        $resultSet9 = $finder->find(new Query('Gallery'));
        $this->assertEquals(
            'Или что-то может называться словом <b>Gallery</b>.',
            $resultSet9->getItems()[0]->getSnippet()
        );

        $resultSet9 = $finder->find(new Query('legacy'));
        $this->assertEquals(
            'Some <b>legacy</b>. To be continued...',
            $resultSet9->getItems()[0]->getHighlightedTitle($stemmer)
        );
```

**Comment by Pathologic** [comment]
it looks like some changes were not merged 
https://github.com/parpalak/rose/commit/a9acae77ce267db6d11270bbf98b4ac190076e0a is absent in 
https://github.com/parpalak/rose/blob/v1.1/src/S2/Rose/Stemmer/PorterStemmerEnglish.php

**Comment by parpalak** [comment]
Oh, https://github.com/parpalak/rose/releases/tag/v1.1 does not contain the last changes intentionally. v1.1 contains the changes that was not released after v1.0. The changes in https://github.com/parpalak/rose/issues/24 break the backward compatibility of some interfaces so I'll release v2.0 with all last changes. Right now I'm testing it and adjusting relevance settings. For now you can take 'master' branch.

**Comment by Pathologic** [comment]
Everything works, thank you again.
